src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -4848,14 +4848,25 @@ export class StudyViewPageStore {
         default: [],
     });
 
+    readonly molecularProfileSamplesSet = remoteData<{
```

Comment:
i think we should improve the name of this and also the typing.    this is a problem in a lot of places but the name doesn't tell me what it is.     like, sampleIdsByMolecularProfileIdSet.   Also, is there a reason we don't just stick with samples themselves, e.g.     [profileid] : Sample[]?    I mean the only reason I can see is performance.

Reply:
yes, that's the only reason. And I think `sampleUniqueKeysByMolecularProfileIdSet`  would be more describing